### DISCUSSION TITLE
Storybook stories escaping quotes

### DISCUSSION THREAD

**Discussion by dgibson666** [storybookjs/storybook#25085] [Help]
### Summary

I'm using Storybook for Vite/HTML showcasing web components built with Stencil. I have sx props on many of my web components, similar to Material React components, which accepts an object (or stringified JSON) of style property/value pairs. It seems that Storybook is escaping quotes, which is breaking this.

In the Storybook HTML tab, it shows the desired rendering of sx='{"color":"red"}'. Since this represents a stringified JSON, the quotes are inverted.

But in the browser DevTools, it shows the inner double quotes escaped, and therefore isn't working correctly because it can't be parsed as JSON. This is happening even if I enter the value as an object control in Storybook. Since I'm using the HTML version of Storybook, it's still stringified into an HTML attribute, which should work.

If I copy the desired output in the first case into the DOM via DevTools, it then works. So it seems like Storybook is escaping the quotes at some point. Does anyone know about this/have a workaround? I didn't have this problem on my previous project that used JSX for stories (custom parser).

**Comment by break-stuff** [comment]
Is this only happening when you are adding values to the control or is it also happening when you add it to a story's args?

**Reply by dgibson666** [reply]
All of the above. In a story where sx is an arg and control, I'm outputting it to an attribute like this inside a template literal story:

```
return ` 
      <my-typography
        ${tag ? `tag=${tag}` : ''}
        ${variant!='none' ? `variant=${variant}` : ''}
        ${divider!='none' ? `divider=${divider}` : ''}
        ${sx ? `sx=${JSON.stringify(sx)}` : ''}
      >
        ${text}
      </my-typography>
    `
```

I thought JSON.stringify might be the culprit, but 1) if I console.log both sx and JSON.stringify(sx), they are as expected (no escaping).

2) I have a snippet as part of another story like this (no JSON.stringify) that does the same thing:

```
<my-typography tag="h6" variant="heading-md" sx='{"margin-bottom":"2rem"}'>Some Text</my-typography>
```

It's not applying the margins because the double quotes are escaped in the DOM/DevTools. If I copy this directly into the DevTools, it then works (if I make it reactive in the component).

**Reply by break-stuff** [reply]
Have you tried using Storybook's web component version? It uses Lit's tagged template literals in stories that may help in this situation.

**Reply by dgibson666** [reply]
No, that's more of a heavy lift. I was hoping this was a known issue and someone had a suggestion. I'm not using Lit to build my components, so I didn't really think it made sense to use that any more than the React version. 

I've always used the HTML version with my Stencil components, per a couple of articles I've read about it (one from Ionic, who makes Stencil). But this is the first project using template literals instead of JSX (I don't have access to a JSX plugin on this project), which is why I've never run into this before.

**Reply by break-stuff** [reply]
Oh, that's a common misconception. You don't have to build your components with Lit to use the web components version of Storybook. It just allows you to use Lit in the templates to provide better data binding and tooling. Since you are already using ES templates, updating your stories would be as simple as adding the `html` prefix to make it a tagged template literal.

**Comment by dgibson666** [comment]
So this turns out to be my error. I was working on another component and the JSON attribute worked just fine. So I did a deep dive into troubleshooting and logging around the other error and it turned out I was generating an error in a specific instance that was not clear/obvious. I found out why it was breaking and fixed it. It appears that JSON.parse knows how to unescape those escaped quotes just fine actually, so no need to switch versions of Storybook.

Thanks for your time and responses!
